index.bs

```diff
@@ -5129,6 +5130,56 @@ if the [=authenticator=] does not support [=attestation=].
 : Verification procedure
 :: Return implementation-specific values representing [=attestation type=] [=None=] and an empty [=attestation trust path=].
 
+## Apple Anonymous Attestation Statement Format ## {#sctn-apple-anonymous-attestation}
+
+This attestation statement format is exclusively used by Apple for certain types of Apple devices that support WebAuthn.
+
+: Attestation statement format identifier
+:: apple
+
+: Attestation types supported
+:: [=Anonymous CA=]
+
+: Syntax
+:: The syntax of an Apple attestation statement is defined as follows:
+
+    ```
+    $$attStmtType //= (
+                          fmt: "apple",
+                          attStmt: appleStmtFormat
+                      )
+
+    appleStmtFormat = {
+                          x5c: [ credCert: bytes, * (caCert: bytes) ]
+                      }
+    ```
+
+    The semantics of the above fields are as follows:
+
+    : x5c
+    :: |credCert| followed by its certificate chain, each encoded in X.509 format.
+
+    : credCert
+    :: The credential public key certificate used for attestation, encoded in X.509 format.
+
+: Signing procedure
+::
+    1. Let |authenticatorData| denote the authenticator data for the attestation, and let |clientDataHash| denote the hash of the serialized client data.
+    2. Concatenate |authenticatorData| and |clientDataHash| to form |nonceToHash|.
+    3. Perform SHA-256 hash of |nonceToHash| to produce |nonce|.
+    4. Let Apple anonymous attestation CA generate a certificate for the credential public key and include the |nonce| as an extension with OID ( 1.2.840.113635.100.8.2 ). |credCert| denotes this certificate.
+    5. Set |x5c| to |credCert| followed by its certificate chain.
+
+: Verification procedure
+:: Given the verification procedure inputs |attStmt|, |authenticatorData| and |clientDataHash|, the verification procedure is as follows:
+    1. Verify that |attStmt| is valid CBOR conforming to the syntax defined above and perform CBOR decoding on it to extract the contained fields.
+    2. Verify |x5c| is a valid certificate chain starting from the |credCert| to the Apple WebAuthn root certificate.
```

Comment:
Would it be possible to include a link to where the Apple WebAuthn root certificate can be downloaded?

Reply:
It seems this is the URL to the root cert: https://www.apple.com/certificateauthority/Apple_WebAuthn_Root_CA.pem (as linked from this page I stumbled upon: https://www.apple.com/certificateauthority/private/)

Reply:
Maybe we could do the same for the safety and android key?

Reply:
The spec seems to have been intentionally written to be non-prescriptive about this. All other attestation formats flow like this:
- [Section 7.1 step 14](https://www.w3.org/TR/webauthn-1/#registering-a-new-credential) kicks things off:
  > Verify that _attStmt_ is a correct attestation statement, conveying a valid attestation signature, by using the attestation statement format _fmt_’s verification procedure given _attStmt_, _authData_ and the hash of the serialized client data computed in step 7.
- In [section 8](https://www.w3.org/TR/webauthn-1/#defined-attestation-formats), find the section of the attestation format in question. In general the final step of the happy path looks something like this:
  > If successful, return implementation-specific values representing attestation type <type> and attestation trust path _x5c_.
- Section 7.1 step 15 then continues with the above, but this makes finding the root of trust generic instead of format specific: 
  > If validation is successful, obtain a list of acceptable trust anchors (attestation root certificates or ECDAA-Issuer public keys) for that attestation type and attestation statement format _fmt_, from a trusted source or from policy. For example, the FIDO Metadata Service provides one way to obtain such information, using the `aaguid` in the `attestedCredentialData` in authData.

Initially I found this annoying as an implementer, the MDS was pretty lacking and actually getting the stuff I needed was hard. Getting TPM root certificates anyone? However since I've come to realize it was probably written this way to allow for:
- vendors choosing not to publish to the Fido MDS for whatever reason (although this would make my life a lot easier if they do)
- the [Fido conformance tool](https://fidoalliance.org/certification/functional-certification/conformance/) to provide their own root certificates for testing purposes, this can't ship with the real Apple cert for obvious reasons
- other less obvious use cases I can't think of right now

So this line should be changed to follow the rest of the spec (or less likely: the rest of the spec should be changed) IMO.

Reply:
Hmm, in that case perhaps the WACG can take this on and maintain links to the various root certs in our "how to WebAuthn" website. It'd allow the spec to remain implementation-agnostic while giving devs one place to go to for links to the latest root certs 🤔

Reply:
> Would it be possible to include a link to where the Apple WebAuthn root certificate can be downloaded?

I do not think that we'd place that sort of link directly in the spec.

> in that case perhaps the WACG can take this on and maintain links to the various root certs in our "how to WebAuthn" website.

Yes, doing something like that is much more plausible.

Reply:
> Would it be possible to include a link to where the Apple WebAuthn root certificate can be downloaded?

I do not think that we'd place that sort of link directly in the spec.

> in that case perhaps the WACG can take this on and maintain links to the various root certs in our "how to WebAuthn" website.

Yes, doing something like that is much more plausible.